paddlenlp/trainer/trainer.py

```diff
@@ -1259,19 +1259,27 @@ def _maybe_log_save_evaluate(self, tr_loss, model, epoch, ignore_keys_for_eval,
             logs["learning_rate"] = float("{0:.3e}".format(self._get_learning_rate()))
             logs["global_step"] = int(self.state.global_step)
 
-            divisor = 2**30
-            # TODO(@gexiao): replace these codes with unified APIs in Paddle
-            current_device = framework._current_expected_place_()
-            if str(current_device) != "Place(cpu)":
-                device_id = current_device.get_device_id()
-                current_memory_allocated = core.device_memory_stat_current_value("Allocated", device_id)
-                current_memory_reserved = core.device_memory_stat_current_value("Reserved", device_id)
-                max_memory_allocated = core.device_memory_stat_peak_value("Allocated", device_id)
-                max_memory_reserved = core.device_memory_stat_peak_value("Reserved", device_id)
-                logs["current_memory_allocated"] = current_memory_allocated / divisor
-                logs["current_memory_reserved"] = current_memory_reserved / divisor
-                logs["max_memory_allocated"] = max_memory_allocated / divisor
-                logs["max_memory_reserved"] = max_memory_reserved / divisor
+            # Add additional memory in log.
+            if not self.args.skip_memory_metrics:
+                divisor = 2**30
+                logs.update(
+                    {
+                        "cpu_mem_used": self._memory_tracker.cpu_mem_used() / divisor,
+                        "cpu_mem_used_peak": self._memory_tracker.cpu_mem_used_peak / divisor,
+                    }
+                )
+                # TODO(@gexiao): replace these codes with unified APIs in Paddle
+                current_device = framework._current_expected_place_()
+                if str(current_device) != "Place(cpu)":
+                    device_id = current_device.get_device_id()
+                    current_memory_allocated = core.device_memory_stat_current_value("Allocated", device_id)
+                    current_memory_reserved = core.device_memory_stat_current_value("Reserved", device_id)
+                    max_memory_allocated = core.device_memory_stat_peak_value("Allocated", device_id)
+                    max_memory_reserved = core.device_memory_stat_peak_value("Reserved", device_id)
+                    logs["current_memory_allocated"] = current_memory_allocated / divisor
+                    logs["current_memory_reserved"] = current_memory_reserved / divisor
+                    logs["max_memory_allocated"] = max_memory_allocated / divisor
```

Comment:
```suggestion
                    logs["max_memory_allocated"] = max_memory_allocated >> 20
```
这个之前是MB单位，建议不要改变了，保持原来写法。用除法的话，是浮点数，还有小数位的问题，建议直接位运算，MB为单位

Reply:
已修改